### PR TITLE
Add filter for inputs

### PR DESCRIPTION
**Pull request by kmerz**
## Description
Add a filter input for inputs to filter inputs on the inputs list page.

## Motivation and Context
If a user has lots of inputs finding the right one can be tiresome with the browser search function.

## How Has This Been Tested?
Add local inputs and filtered them.

## Screenshots (if appropriate):
screenshot-2018-5-24 graylog - inputs 1

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

Fixes: #2403

### CODE REVIEW

**Comment by kmerz** [comment]
@edmundoa Yes you are right, the ```SearchForm``` does what I need, I do not know why I did not take it in the first place even though I found it.

**Comment by edmundoa** [comment]
@kmerz it's hard to find all components that we have, the frontend documentation helps a little, but not sure how much. We also lack some information regarding where to use certain components or others. I guess that would be something to add in the component gallery, too.

File: graylog2-web-interface/src/components/inputs/InputsList.jsx [hunk 1]
```diff
@@ -83,15 +136,28 @@ const InputsList = createReactClass({
...
 
+        <Row id="filter-input" className="content input-new">
+          <Col>
+            <SearchForm onSearch={this._onFilterInputs}
+                        label="Filter inputs"
+                        topMargin={0}
+                        onReset={this._onFilterReset}
+                        wrapperClass={InputsListStyle.filterWrapper}
+                        searchButtonLabel="Filter"
+                        placeholder="Search by title"
```
Comment (edmundoa) [inline graylog2-web-interface/src/components/inputs/InputsList.jsx]: I think having an input above without any label and another one below it with a label looks kind of funny. Maybe should we try to make both inputs look similar? This is a quick attempt:

Before:
before

After:
after

Comment (edmundoa) [reply graylog2-web-interface/src/components/inputs/InputsList.jsx]: In any case we should only use one verb in the `SearchForm`, either "search" or "filter".

File: graylog2-web-interface/src/components/inputs/InputsList.jsx [hunk 2]
```diff
@@ -55,7 +65,12 @@ const InputsList = createReactClass({
       localInputs = localInputs.filter(input => input.node === this.props.node.node_id);
     }
 
-    this.setState({ globalInputs: globalInputs, localInputs: localInputs });
+    this.setState({
+      globalInputs: globalInputs,
+      filteredGlobalInputs: globalInputs,
+      localInputs: localInputs,
+      filteredLocalInputs: localInputs,
+    });
```
Comment (edmundoa) [inline graylog2-web-interface/src/components/inputs/InputsList.jsx]: We also need to update the filtered inputs when props change. For instance, when a new input is added, we should keep the filter or at least reset it. Right now the filter seems to be in place, but the list is not filtered.

Comment (edmundoa) [reply graylog2-web-interface/src/components/inputs/InputsList.jsx]: Sorry, I meant when the state propagated from the store changes :)

File: graylog2-web-interface/src/components/inputs/InputsList.jsx [hunk 3]
```diff
@@ -83,26 +144,33 @@ const InputsList = createReactClass({
...
+        <Row id="filter-input" className="content input-list">
           <Col md={12}>
-            <h2>
+            <SearchForm onSearch={this._onFilterInputs}
+                        topMargin={0}
+                        onReset={this._onFilterReset}
+                        searchButtonLabel="Filter"
+                        placeholder="Filter by title" />
+            <br />
+            <h1>
```
Comment (edmundoa) [inline graylog2-web-interface/src/components/inputs/InputsList.jsx]: Looking at this again, `h1` feels too big. Could we make it `h2` again and increase the spacing between global and local inputs? I think that could make the section a bit less prominent, while keeping some separation between them.

File: graylog2-web-interface/src/components/inputs/InputsList.jsx [hunk 4]
```diff
@@ -70,6 +85,52 @@ const InputsList = createReactClass({
...
+    }
+
+    const filterMethod = (input) => {
+      return regExp.test(input.title);
+    };
+    const filteredGlobalInputs = this.state.globalInputs.filter(filterMethod);
+    const filteredLocalInputs = this.state.localInputs.filter(filterMethod);
+    this.setState({
+      filteredGlobalInputs: filteredGlobalInputs,
+      filteredLocalInputs: filteredLocalInputs,
```
Comment (edmundoa) [inline graylog2-web-interface/src/components/inputs/InputsList.jsx]: I think we are missing to store the `filter` in the state in here.

File: graylog2-web-interface/src/components/rules/RuleHelper.jsx [hunk 5]
```diff
@@ -170,7 +179,11 @@ end`,
...
                 <Row>
                   <Col sm={12}>
-                    <FilterInput onChange={this._filterDescriptors} />
+                    <SearchForm onSearch={this._filterDescriptors}
+                                label="Filter rules"
+                                topMargin={0}
+                                searchButtonLabel="Filter"
+                                onReset={this._onFilterReset} />
                     <div className={`table-responsive ${RuleHelperStyle.marginTab}`}>
                       <PaginatedList totalItems={functionDescriptors.length} pageSize={this.state.pageSize} onChange={this._onPageChange} showPageSizeSelect={false}>
```
Comment (edmundoa) [inline graylog2-web-interface/src/components/rules/RuleHelper.jsx]: I don't think is very important, or if it's related to this change, but the save/reset page is not working as I expected. Examples:

1. Go to page 3 of functions
2. Filter `is`, as that's the function I want to use
3. The pagination says I'm in page 3, but I am actually in page 1

--

1. Filter items with `is`
2. Go to page 2
3. Reset the filter
4. The pagination says I'm in page 2, but I'm in page 1

---

After looking at the code a bit, I think all we are missing is to pass the `activePage` prop to `PaginatedList`. That should make the pager change the page when changed externally.

**Comment by edmundoa** [comment]
LGTM 👍

**Pull request merged**